qiime2/core/type/collection.py

```diff
@@ -95,6 +109,28 @@ def validate_field(self, name, field):
         pass
 
 
+class _Collection(_1DCollectionBase):
+    _view = dict
+
+    def is_element_expr(self, self_expr, value):
+        """Since this is a dictionary, we often need to make sure to use its
+        values not its keys.
+        """
+        contained_expr = self_expr.fields[0]
+        if isinstance(value, self._view) and len(value) > 0:
+            return all(v in contained_expr for v in value.values())
+        # The Collection's default view type is dict, but it is also allowed to
+        # be passed around as a list
+        elif len(value) > 0:
+            return all(v in contained_expr for v in value)
+
+        return False
+
+    def __iter__(self):
+        return self._view.values()
```

Comment:
```suggestion
```

`__iter__` is very special to the type system, so I would avoid implementing it here